nwbwidgets/ophys.py

```diff
@@ -113,52 +112,120 @@ def show_plane_segmentation_2d(plane_seg: PlaneSegmentation, color_wheel=color_w
-                               isinstance(plane_seg[col][i], (int, float, np.integer, np.float, str))])
-        hovertext += '<br>' + sec_str
-        # form cell borders
-        x, y = compute_outline(plane_seg['image_mask'][i], threshold)
-
-        fig.add_trace(
-            go.Scatter(
-                x=x, y=y,
-                fill='toself',
-                mode='lines',
-                name=str(plane_seg[color_by][i]),
-                legendgroup=str(plane_seg[color_by][i]),
-                showlegend=show_leg,
-                text=hovertext,
-                hovertext='text',
-                line=dict(width=.5),
-                **kwargs
+        aux_leg = []
+        for i in range(nUnits):
+            if plane_seg[color_by][i] not in aux_leg:
+                show_leg = True
+                aux_leg.append(plane_seg[color_by][i])
+            else:
+                show_leg = False
+            kwargs = dict()
+            if color_by:
+                c = color_wheel[np.where(cats == plane_seg[color_by][i])[0][0]]
+                kwargs.update(line_color=c)
+            # hover text
+            hovertext = '<b>roi_id</b>: ' + str(plane_seg.id[i])
+            rois_cols = list(plane_seg.colnames)
+            if 'roi_id' in rois_cols:
+                rois_cols.remove('roi_id')
+            sec_str = '<br>'.join([col + ': ' + str(plane_seg[col][i]) for col in rois_cols if
+                                   isinstance(plane_seg[col][i], (int, float, np.integer, np.float, str))])
+            hovertext += '<br>' + sec_str
+            # form cell borders
+            x, y = compute_outline(plane_seg['image_mask'][i], threshold)
+
+            fig.add_trace(
+                go.Scatter(
+                    x=x, y=y,
+                    fill='toself',
+                    mode='lines',
+                    name=str(plane_seg[color_by][i]),
+                    legendgroup=str(plane_seg[color_by][i]),
+                    showlegend=show_leg,
+                    text=hovertext,
+                    hovertext='text',
+                    line=dict(width=.5),
+                    **kwargs
+                )
+            )
+            fig.update_layout(
+                width=700, height=500,
+                margin=go.layout.Margin(l=60, r=60, b=60, t=60, pad=1),
+                plot_bgcolor="rgb(245, 245, 245)",
             )
-        )
-        fig.update_layout(
-            width=700, height=500,
-            margin=go.layout.Margin(l=60, r=60, b=60, t=60, pad=1),
-            plot_bgcolor="rgb(245, 245, 245)",
-        )
+    else:
+        fig.update_traces(visible=False)
+        for i in range(len(fig.data)):
+            fig.data[i].line.color = color_wheel[np.where(cats == plane_seg[color_by][i])[0][0]]
+        fig.update_traces(visible=True)
+        """kwargs.update_traces()
+Here change the traces?
+"""
+        # fig.data = None
+
     return fig
 
 
+"""NEW CLASS"""
```

Comment:
remove this comment. Comments are good, but they should reflect what the code does, not how you develop. If you want to describe your process, git commit messages are a good place to put that